code/src/main.cpp

```diff
@@ -1,9 +1,11 @@
-#include <iostream>
-
-#include "rtweekend.h"
-#include "hittablelist.h"
 #include "color.h"
+#include "hittablelist.h"
+#include "ray.h"
 #include "sphere.h"
+#include "utils.h"
+#include "vector3.h"
+
+#include <iostream>
```

Comment:
Accroding to the Google Standard the include headers should be in this oder order: 
Related header, C system headers, C++ standard library headers, other libraries' headers, your project's headers.
 https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

Reply:
You're right! But I checked a couple of questions on StackOverflow that argue that style will only hide any missing dependencies in your modules. For example, if you have:

**my_module.h**
```
#pragma once

int DoSomething(std::string some_text) {
// code here
}
```

**main.cpp**
```
#include <string>
#include "my_module.h"

int main() {
// code here
}
```

This will compile, but the order of the `include`s will hide the fact that `my_module.h` does not include all the headers it needs, i.e. `<string>`.

The rules I'm suggesting come from [this answer on SO](https://stackoverflow.com/a/2762596/4160025). [Another answer](https://stackoverflow.com/a/14243688/4160025) specifically addresses the issue with Google's style in regards to the order of `include`s. [Someone asked about this style](https://stackoverflow.com/questions/43420799/why-does-the-google-style-guide-dictates-that-system-headers-come-before-project) and what I get from the discussion is that it'll be easier to catch these dependency bugs if we make it hard to hide them.

Reply:
Google conventions have done it again hahahah
This makes perfect sense, and since you've already added it to our docs I guess we are good to go, pleade do include those links into our discord server in "good practices'.

Reply:
Will do! :)